src/cli.c

```diff
@@ -1406,3 +1422,287 @@ process_result(SSL* ssl, int socket, int32_t output_format)
+   char* translated_compression = NULL;
+   char* translated_encryption = NULL;
+
+   translated_valid = translate_valid((int32_t)pgmoneta_json_get(response, MANAGEMENT_ARGUMENT_VALID));
+   if (translated_valid)
+   {
+      pgmoneta_json_put(response, MANAGEMENT_ARGUMENT_VALID, (uintptr_t)translated_valid, ValueString);
+   }
+   translated_compression = translate_compression((int32_t)pgmoneta_json_get(response, MANAGEMENT_ARGUMENT_COMPRESSION));
+   if (translated_compression)
+   {
+      pgmoneta_json_put(response, MANAGEMENT_ARGUMENT_COMPRESSION, (uintptr_t)translated_compression, ValueString);
+   }
+   translated_encryption = translate_encryption((int32_t)pgmoneta_json_get(response, MANAGEMENT_ARGUMENT_ENCRYPTION));
+   if (translated_encryption)
+   {
+      pgmoneta_json_put(response, MANAGEMENT_ARGUMENT_ENCRYPTION, (uintptr_t)translated_encryption, ValueString);
+   }
+
+   free(translated_valid);
+   free(translated_compression);
+   free(translated_encryption);
+}
+
+static void
+translate_json_object(struct json* j)
+{
+   struct json* header = NULL;
+   int32_t command = 0;
+   char* translated_command = NULL;
+   int32_t out_format = -1;
+   char* translated_out_format = NULL;
+   struct json* response = NULL;
+
+   struct json* backups = NULL;
+   struct json* backup = NULL;
+   struct deque* backup_dq = NULL;
+   struct deque_iterator* backup_dq_it = NULL;
+   struct json* servers = NULL;
+   struct deque* server_dq = NULL;
+   struct deque_iterator* server_dq_it = NULL;
+
+   // Translate arguments of header
+   header = (struct json*)pgmoneta_json_get(j, MANAGEMENT_CATEGORY_HEADER);
+
+   if (header)
+   {
+      command = (int32_t)pgmoneta_json_get(header, MANAGEMENT_ARGUMENT_COMMAND);
+      translated_command = translate_command(command);
+      if (translated_command)
+      {
+         pgmoneta_json_put(header, MANAGEMENT_ARGUMENT_COMMAND, (uintptr_t)translated_command, ValueString);
+      }
+
+      out_format = (int32_t)pgmoneta_json_get(header, MANAGEMENT_ARGUMENT_OUTPUT);
+      translated_out_format = translate_output_format(out_format);
+      if (translated_out_format)
+      {
+         pgmoneta_json_put(header, MANAGEMENT_ARGUMENT_OUTPUT, (uintptr_t)translated_out_format, ValueString);
+      }
+
+      free(translated_command);
+      free(translated_out_format);
+   }
+
+   // Translate the response
+   response = (struct json*)pgmoneta_json_get(j, MANAGEMENT_CATEGORY_RESPONSE);
+
+   if (response && command)
+   {
+      switch (command)
+      {
+         case MANAGEMENT_BACKUP:
+         case MANAGEMENT_RESTORE:
+         case MANAGEMENT_RETAIN:
+         case MANAGEMENT_EXPUNGE:
+         case MANAGEMENT_INFO:
+         case MANAGEMENT_ANNOTATE:
+            translate_backup_argument(response);
+            break;
+         case MANAGEMENT_LIST_BACKUP:
+            backups = (struct json*)pgmoneta_json_get(response, MANAGEMENT_ARGUMENT_BACKUPS);
+            backup_dq = (struct deque*)backups->elements;
```

Comment:
We have a json wrapper for the internal deque. Use json_iterator directly: https://github.com/pgmoneta/pgmoneta/blob/0476e50f8c0d1da9bf87ba18050af5a654684560/src/include/json.h#L193

Reply:
It has never been really put to use before, so let us know if it doesn't work (that would be my fault).

Reply:
@Jubilee101 PTAL